### DISCUSSION TITLE
Sparse matrix deep copy

### DISCUSSION THREAD

**Discussion by ArmoredCat** [optimatika/ojAlgo#558] [General, ojAlgo and related code]
Is there currently a way to get a deep copy of a SparseStore instance?

**Comment by ArmoredCat** [comment]
Current way I am doing it is by filling the diagonals of a new matrix with 1.0 and then multiplying with the matrix you want to copy, but there probably is a more efficient way. Also tried the nonzero() method and just setting it manually but that takes the same amount of time surprisingly.

**Comment by apete** [comment]
I'd say using `nonzeros()` is the best option. A recent change/addition (have to build and use a snapshot version) allow to specify the initial capacity (number of nonzeros) of the receiver. That can make a big difference.

**Reply by ArmoredCat** [reply]
That worked great, slashed my runtime completely, thank you!

**Reply by apete** [reply]
If you built a snapshot version from the develop branch, then there is now a `copy` function in the factory class. In addition to what (I assume) you have now this also bypasses some internally `synchronized` code. Should speed things up further.

https://github.com/optimatika/ojAlgo/commit/bdf82c0bb5ba0e46a4fbd3a9f2aeb779750ae70f
